components/layout_2020/flow/mod.rs

```diff
@@ -141,18 +141,20 @@ impl BlockLevelBox {
             .auto_is(|| {
                 let margin_inline_start = pbm.margin.inline_start.auto_is(Length::zero);
                 let margin_inline_end = pbm.margin.inline_end.auto_is(Length::zero);
-                containing_block.inline_size -
-                    pbm.padding_border_sums.inline.into() -
-                    margin_inline_start -
-                    margin_inline_end
+                Length::from(
+                    containing_block.inline_size -
+                        pbm.padding_border_sums.inline -
+                        margin_inline_start.into() -
+                        margin_inline_end.into(),
+                )
```

Comment:
Can avoid the `into()`:
```suggestion
                Length::from(containing_block.inline_size - pbm.padding_border_sums.inline) -
                    margin_inline_start -
                    margin_inline_end
```